ui/src/utils/throttleState.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+
+export function useThrottleState<T>(
+  current: T,
+  updateBackingValue: (newValue: T) => void,
+  {
+    timeout = 300,
+    immediatelyUpdateIfFalsey = false,
+  }: { timeout?: number; immediatelyUpdateIfFalsey?: boolean } = {}
+): [T, (newValue: T) => void] {
+  const [value, setValue] = React.useState(current);
```

Comment:
Shoutout to Github Copilot, which filled in the entire function at this point and was completely wrong about what I wanted it to do 🤣